ifsbench/data/tests/test_fetchhandler.py

```diff
@@ -0,0 +1,126 @@
+# (C) Copyright 2020- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+"""
+Tests for the FetchHandler class.
+"""
+
+from pathlib import Path
+
+import pytest
+
+from ifsbench.data import FetchHandler
+
+
+@pytest.mark.parametrize('source_url', ['file://home/some_file.tar.gz'])
+@pytest.mark.parametrize('target_path', [Path('subdir/target.tar.gz')])
+@pytest.mark.parametrize('force', [True, False])
+@pytest.mark.parametrize('ignore_errors', [True, False])
+def test_fetchhandler_serialise(source_url, target_path, force, ignore_errors):
+    """
+    Serialise and deserialise a FetchHandler and make sure the the resulting
+    object is correct.
+    """
+    config = {
+        'source_url': source_url, 
+        'target_path': target_path,
+        'force': force,
+        'ignore_errors': ignore_errors
+    }
+
+    handler = FetchHandler(**config)
+
+    config_dump = handler.dump_config()
+    new_handler = FetchHandler.from_config(config_dump)
+
+    assert new_handler.source_url == source_url
+    assert new_handler.target_path == target_path
+    assert new_handler.force == force
+    assert new_handler.ignore_errors == ignore_errors
+
+@pytest.fixture(name='run_dir')
+def fixture_rundir(tmp_path):
+    """
+    Create temporary directory with two files.
+    """
+    with (tmp_path/'hello.txt').open('w') as f:
+        f.write("Hello!")
+    with (tmp_path/'goodbye.txt').open('w') as f:
+        f.write("Goodbye!")
+
+    return tmp_path
+
+@pytest.mark.parametrize('force', [True, False])
+@pytest.mark.parametrize('ignore_errors', [True, False])
+def test_fetchhandler_force(run_dir, force, ignore_errors):
```

Comment:
I think this is better to split into two. In code we want to reduce duplication, in tests it's better to be more verbose (DAMP not DRY).